src/core/thread/key_manager.cpp

```diff
@@ -396,16 +509,41 @@ void KeyManager::IncrementMleFrameCounter(void)
 
 void KeyManager::SetKek(const Kek &aKek)
 {
-    mKek             = aKek;
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    OT_ASSERT(ImportKek(aKek.GetKey(), aKek.kSize) == kErrorNone);
```

Comment:
Yes, I was aware of this. But chose to do it anyway s as to help development activity. `ImportKek` should not fail unless the PSA is not configured correctly.

Reply:
If assert is disabled, we will not even call  `ImportKek()` method. I don't think that is what we want, is it?

The point is that if assert is disabled (`OPENTHREAD_CONFIG_ASSERT_ENABLE` is `0`) then the `OT_ASSERT(statement)` does not call or perform the `statement` at all. The whole `OT_ASSERT(statement)` line is removed since `OT_ASSERT()` is defined as an empty macro. Here is the simplified related code in `debug.hpp`
```c++
#if OPENTHREAD_CONFIG_ASSERT_ENABLE
...
#else 
#define OT_ASSERT(cond)
#endif 
```